plugins/legacy-preset-chart-nvd3/src/NVD3Vis.js

```diff
@@ -309,10 +311,18 @@ function nvd3Vis(element, props) {
   const container = element;
   container.innerHTML = '';
   const activeAnnotationLayers = annotationLayers.filter(layer => layer.show);
-  const chartId =
-    container.parentElement && container.parentElement.id !== ''
-      ? container.parentElement.id
-      : null;
+
+  // Search for the chart id in a parent div from the nvd3 chart
+  let chartContainer = container;
+  let chartId = null;
+  while (chartContainer.parentElement) {
+    if (chartContainer.parentElement.id.startsWith(CHART_ID_PREFIX)) {
```

Comment:
```suggestion
    if ((chartContainer.parentElement.id || '').startsWith(CHART_ID_PREFIX)) {
```

Could `chartContainer.parentElement.id` be `undefined`?

Reply:
lol, when it's not defined it's empty string :P don't you love vanilla js/html?

Reply:
Nice